Adds paddingSize prop to EuiAccordion

snide opened the pull request:
Fixes https://github.com/elastic/eui/issues/695

This adds a new prop to `EuiAccordion` of `paddingSize` and accepts the usual array of named sizes. The issue @pugnascotia noticed happened because if you use a FlexGroup inside of an Accordion the negative margin will conflict with the accordions overflow attributes (which we use to hide the content). This causes a bunch of unnecessary scroll bars. Although I toyed with the idea of removing the negative margins from flexgroups entirely, they do have some really useful abilities that would be tough to replicate (wrapping an unknown amount of flexitems dynamically).

![image](https://user-images.githubusercontent.com/324519/39218650-9a8d17e6-47da-11e8-9cbc-e428a840beb5.png)

![image](https://user-images.githubusercontent.com/324519/39218730-f92cc788-47da-11e8-825c-507ce18b4192.png)

snide commented on src/components/accordion/_accordion_form.scss:
```diff
@@ -1,7 +1,3 @@
-.euiAccordionForm__children {
```
This was never properly componentized. I was using it in the example docs as a singleton class, which is why the forms with flex groups weren't breaking. It is no longer needed now that there's a prop to manage padding.

cchaos commented on src/components/accordion/accordion.js:
```diff
@@ -124,15 +143,38 @@ export class EuiAccordion extends Component {
+  /**
+   * The padding around the exposed accordion content.
+   */
+  paddingSize: PropTypes.string,
```
Should this be the `.oneOf()` type syntax?

cchaos commented on src/components/accordion/_accordion.scss:
```diff
@@ -19,11 +19,26 @@
   overflow-y: hidden;
   transform: translatez(0);
   transition:
-    height $euiAnimSpeedNormal $euiAnimSlightResistance,
```
Oh I see you actually removed it entirely. Are you thinking we'll come up with a different solution?

snide replied:
Good catch, this was accidental!

cchaos commented on src/components/accordion/accordion.js:
```diff
@@ -115,7 +132,9 @@ export class EuiAccordion extends Component {
         >
           <div ref={node => { this.childContent = node; }}>
-            {children}
+            <div className={paddingClass}>
```
Any reason not to put the paddingClass directly on the ref'd div versus creating another dom node layer?

snide replied:
I tried that originally. It didn't work with the height calculations and messed up the calculation.

cchaos commented on src/components/accordion/accordion.js:
```diff
@@ -124,15 +143,38 @@ export class EuiAccordion extends Component {
+   */
   buttonContent: PropTypes.node,
+  /**
+   * Will apear right aligned against the button. Useful for separate actions like deletions.
```
Sp: apear -> appear

snide commented:
Addressed feedback. After chatting a bit with @cchaos we decided to make the `paddingSize` default to "none", which makes this a non-breaking change. There's a callout in the docs that explains the scrolling issue, and gives a super easy remedy (essentially use the prop to add appropriate padding).
